### PR TITLE
Crawler Implementation

### PR DESCRIPTION
**Pull request by lameinthebox**
1. placed _**chromedriver**_
2. added a minor fix to _**crawler.py**_; _**crawler.py**_ is, in fact, redundant now, but I'd like to save it for later use.
3. added uniqueness property for **College**, **Department**, **Major**; If not, our crawler creates plenty of colleges/departments/majors of the same name.
4. implemented _**crawler/crawler.py**_;

### CODE REVIEW

**Comment by holenet** [comment]
Does it work on the remote server envirionment, not local?

File: backend/crawler/crawler.py [hunk 1]
```diff
@@ -6,40 +6,185 @@
 checks cancel_flag of it.
 If cancel_flag is True, it stops current job and set status to 'canceled'.
 """
+import os
+
+dir_path = os.path.dirname(__file__)
+driver_path = os.path.join(dir_path, '../../chromedriver')
```
Comment (holenet) [inline backend/crawler/crawler.py]: I think it is better to get `driver_path` based on `BASE_DIR` of `settings.py`, not based on this file.
https://github.com/SWPP/ttrs-backend/blob/8646f2d124ed11e9c91437b9b63b35799fd2d3e6/backend/backend/settings.py#L15-L16

like this,
```python3
from django.conf import settings

driver_path = os.path.join(settings.BASE_DIR, ...)
```

File: backend/crawler/crawler.py [hunk 2]
```diff
@@ -6,40 +6,185 @@
...
+    driver.find_element_by_xpath('//*[@id="srchOpenShtm"]/option[{}]'.format(sid[semester])).click()
+    driver.find_element_by_xpath('//*[@id="srchOpenSubmattCorsFg"]/option[2]').click()
+    driver.find_element_by_class_name('btn_search_ok').click()
+
+    total_cnt = int(driver.find_element_by_xpath('//*[@id="content"]/div/div[3]/div[1]/div[1]/h3/span').text)
+    total_page = (total_cnt // 10) + 1
 
     try:
-        for i in range(total_cnt):
+        for i in range(1, total_page):
```
Comment (holenet) [inline backend/crawler/crawler.py]: this `for` will iterate `total_page - 1` times.
Please check it is really your intent.

**Comment by lameinthebox** [comment]
Well, it does work well with machines _with display_..I think it is enough for now to build db in dev machines and send it to our server machine..? Isn't it better to further develop _displayless crawler_ when we are finished building major functionalities and start to implement _fancy functions_?

File: backend/crawler/crawler.py [hunk 3]
```diff
@@ -6,40 +6,185 @@
...
+    driver.implicitly_wait(3)
+
+    driver.get('https://sugang.snu.ac.kr/sugang/cc/cc100.action')
+    driver.find_element_by_id('detail_button').click()
+    driver.find_element_by_xpath('//*[@id="srchOpenShtm"]/option[{}]'.format(sid[semester])).click()
+    driver.find_element_by_xpath('//*[@id="srchOpenSubmattCorsFg"]/option[2]').click()
+    driver.find_element_by_class_name('btn_search_ok').click()
+
+    total_cnt = int(driver.find_element_by_xpath('//*[@id="content"]/div/div[3]/div[1]/div[1]/h3/span').text)
+    total_page = (total_cnt // 10) + 1
```
Comment (holenet) [inline backend/crawler/crawler.py]: This is not valid calculation for `total page`.
When `total_cnt` is 100, `total_page` will be `(100//10)+1 = 11` is not equal to actual # of pages, which is 10.

You have to change this calculation to,
```python3
total_page = ((total_cnt-1) // 10 + 1
```

**Review by holenet — CHANGES_REQUESTED** [review]
Please remove `backend/crawler.py`.
This file is no more necessary.

In addition, please do not use one or two character variable name. It lowers the readability.

**Comment by holenet** [comment]
@lameinthebox 
Okay, I think crawling on the server machine will cause some troubles and doing this job later is good enough to now.

**Pull request merged**